spec/generator_spec.rb

```diff
@@ -55,35 +42,80 @@
   end
 end
 
-describe AliquotPay do
-  context :ECv1 do
-    let(:instance)     { AliquotPay.new(:ECv1) }
-    include_examples 'generation tests'
+shared_examples 'ECv1' do
+  it 'has the correct protocolVersion' do
+    expect(instance.token['protocolVersion']).to eq('ECv1')
+  end
 
-    it 'has the correct protocolVersion' do
-      expect(instance.token['protocolVersion']).to eq('ECv1')
+  it 'excludes intermediateSigningKey' do
+    expect(instance.token['intermediateSigningKey']).to be nil
+  end
+
+end
+
+shared_examples 'ECv2' do
+  context 'signed_key' do
+    let(:message) { instance.build_signed_key }
+    let(:result) { Aliquot::Validator::SignedKeyContract.schema.call(message) }
+
+    it 'validates' do
+      expect(result.success?).to be(true), result.errors.to_s
     end
 
-    it 'excludes intermediateSigningKey' do
-      expect(instance.token['intermediateSigningKey']).to be nil
+    it 'has the correct protocolVersion' do
+      expect(instance.token['protocolVersion']).to eq('ECv2')
     end
   end
+end
 
-  context :ECv2 do
-    let(:instance)     { AliquotPay.new(:ECv2) }
-    include_examples 'generation tests'
+describe AliquotPay do
+  context :ECv1 do
```

Comment:
Using a symbol here feels a bit off with the other contexts and shared examples.

Reply:
Or well. I don't favor one or the other, only some kind of consistency.

Reply:
But it's not super important.

Reply:
I agree. It was done to keep the changes to a minimum